### PR TITLE
refactor: extract key/address/tx

### PR DESCRIPTION
**Pull request by classicalliu**
1. extract key / address / transaction as a class
2. move address info out of `API`

### CODE REVIEW

File: lib/ckb/address.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module CKB
+  class Address
+    attr_reader :pubkey
+
+    MODE_TESTNET = "testnet"
+    MODE_MAINNET = "mainnet"
+    MODE_CUSTOM = "custom"
```
Comment (ashchan) [inline lib/ckb/address.rb]: Isn't this used anywhere? If so can we remove it for now?

File: lib/ckb/address.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    def blake160
+      @blake160 ||= self.class.blake160(@pubkey)
+    end
+
+    # Generates address assuming default lock script is used
+    # payload = type(01) | bin-idx("P2PH" => "50/32/50/48") | pubkey blake160
+    # see https://github.com/nervosnetwork/ckb/wiki/Common-Address-Format for more info.
+    def generate
+      blake160_bin = [blake160[2..-1]].pack("H*")
+      payload = ["0150325048"].pack("H*") + blake160_bin
```
Comment (ashchan) [inline lib/ckb/address.rb]: `0150325048` is too magic even with the payload comment. 🤔

File: lib/ckb/address.rb (hunk 1 above)
Comment (classicalliu) [reply lib/ckb/address.rb]: Move to `ckb/mode.rb`

File: lib/ckb/address.rb (hunk 2 above)
Comment (classicalliu) [reply lib/ckb/address.rb]: How about this https://github.com/nervosnetwork/ckb-sdk-ruby/pull/26/files#diff-d1fae9cd48181daa7305586f3950fc2fR29

File: lib/ckb/address.rb (hunk 1 above)
Comment (ashchan) [reply lib/ckb/address.rb]: Good move. But still, is `custom` mode even used? And what does that mean? Connecting to a custom node (URL)? If so, that node is still either testnet (including private network) or mainnet.

Comment (classicalliu) [reply lib/ckb/address.rb]: Yes, it's useless now, removed.

**Pull request merged**